pipeline/setup-runner.py

```diff
@@ -0,0 +1,347 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+#!/usr/bin/env python3
+"""Setup runner for Houdini integration tests in CodeBuild."""
+import argparse
+import hashlib
+import os
+import platform
+import subprocess
+import sys
+import time
+from pathlib import Path
+
+import boto3
+from botocore.config import Config
+
+HOUDINI_VERSIONS = ["19.5.805", "20.0.896", "20.5.613", "21.0.440"]
+
+HOUDINI_CHECKSUMS = {
+    "19.5.805": {
+        "linux": "85b04a6b250a5bdb3f5f229d00f95f84308a3f2a40486799a5dc081f586f8632",
+        "windows": "3244510bcc3caefb509d94f79d61fc22ee05190fa5bb123750c100c31e694530",
+        "macos": "8f4b6d3674f2677bcb93115cfee83076e7d978b35ceb743e31c8cbb1d84104b7",
+    },
+    "20.0.896": {
+        "linux": "a68d3b2788db73774febe4e4cd3b5f5c31dca9e902b60b01c0c14b26d2886cc7",
+        "windows": "e21c1b7ec2e80f83a66fe3c9b00dd5579112414efcf647dfaf407f5a94a392ef",
+        "macos": "054c8457529a95475c6cc5d9377ba153746ecebd85a718c8d95c7fb0f4b1a192",
+    },
+    "20.5.613": {
+        "linux": "5b38cf94fb7a87c9398614caef0b6e4357a0fb656e008b84e715bafb28e18fdf",
+        "windows": "f721fbce0202c21c9bb47fae05aa91559818903c45e703db67da11bb494af88e",
+        "macos": "4243eb9442c8c46d7e1a463d8d5b68163bbe09f34d4c811d3391e5141254b983",
+    },
+    "21.0.440": {
+        "linux": "a87451f9146d52051a9ba142d535936638351526a48cba0c6156a221f3be58e6",
+        "windows": "a78e468e99d1be3476b46062e3a043ecc435751b9a9f92e39bee414ace8ce59e",
+        "macos": "3fc918428b22b3c32704d1163a6b1b08723ece71e5e0172e941167549d4d5b25",
+    },
+}
+
+
+def run(cmd, check=True):
+    print(f"Running: {cmd if isinstance(cmd, str) else ' '.join(cmd)}")
```

Comment:
nit: could use [`shlex.join(cmd)`](https://docs.python.org/3/library/shlex.html#shlex.join) for a more correct shell command being logged.